src/overmap_ui.cpp

```diff
@@ -645,6 +646,8 @@ static void draw_ascii(
         size_t count = 0;
     };
     std::unordered_set<tripoint_abs_omt> npc_path_route;
+    std::unordered_set<tripoint_abs_omt> newly_revealed( player_character.map_revealed_omts.begin(),
```

Comment:
You are only using `newly_revealed` variable inside of `if` statement below, so it could be put there and it won't populate it each time even if overmap uistate does not imply showing map revealed omts.